### PR TITLE
IRIS-2868 - add privelaged link to Special:DiscussionsLog in user activity page

### PR DESCRIPTION
**Pull request by katanka**
## Links

* http://wikia-inc.atlassian.net/browse/IRIS-2868

## Description

This adds a link to Special:DiscussionsLog in the user activity page. Additionally, it generalized the `discussion-days-toggle` to a generic `discussion-dropdown`, which will display items that are passed into it.

## Reviewers

Use `@` mentions here. Alternatively, assign the pull request to a person.

### CODE REVIEW

File: front/main/app/templates/components/discussion-community-unit.hbs [hunk 1]
```diff
@@ -28,7 +28,22 @@
 	</a>
 {{/if}}
 
-{{discussion-moderator-tools}}
+{{#discussion-moderator-tools}}
+	<li>
+		{{#link-to 'discussion.reported-posts' class='text-link active-element-theme-color'
+		click=params.trackReportedContentTap}}
```
Comment (jsutterfield) [inline front/main/app/templates/components/discussion-community-unit.hbs]: Super minor: The formatting on this line and L41 looks a little off. Could we indent those some? Maybe to be in line with `discussion.reported-posts`?

File: front/main/app/templates/components/discussion-filters.hbs [hunk 2]
```diff
@@ -28,7 +28,22 @@
 			}}
 		{{/if}}
 
-		{{discussion-moderator-tools}}
+		{{#discussion-moderator-tools}}
+			<li>
+				{{#link-to 'discussion.reported-posts' class='text-link active-element-theme-color'
+				click=params.trackReportedContentTap}}
```
Comment (jsutterfield) [inline front/main/app/templates/components/discussion-filters.hbs]: Some comment about the formatting here.

File: front/main/app/templates/components/discussion-moderator-tools.hbs [hunk 3]
```diff
@@ -3,19 +3,9 @@
...
-		</li>
-		<li>
-			{{#link-to 'discussion.moderator.user-activity' (query-params days="30")
-					   class='text-link active-element-theme-color' click=(action 'trackUserActivityTap')}}
-				{{svg 'user-activity' role='img' class='icon user-activity'}}
-				<span>{{i18n 'main.user-activity' ns='discussion'}}</span>
-			{{/link-to}}
-		</li>
+		{{yield (hash
+		trackReportedContentTap=(action 'trackReportedContentTap')
```
Comment (jsutterfield) [inline front/main/app/templates/components/discussion-moderator-tools.hbs]: We could probably benefit from some more indentation here too.

File: front/main/app/components/application-wrapper.js [hunk 4]
```diff
@@ -42,6 +42,11 @@ export default Component.extend(ResponsiveMixin, {
 		return `${vertical}-vertical`;
 	}),
 
+	viewChangeObserver: observer('responsive.isMobile', function () {
+		// Deregister click handlers for open discussion-dropdown components
```
Comment (jsutterfield) [inline front/main/app/components/application-wrapper.js]: When I first saw this comment, the word "Deregister" seemed off too me. I was going to suggest "unregister". Then I read [this answer](https://english.stackexchange.com/questions/25931/unregister-vs-deregister/40095#40095) and it changed my mind. Carry on then :)

File: front/common/public/locales/en/discussion.json [hunk 5]
```diff
@@ -149,7 +149,8 @@
     "user-activity": "Discussions Insights",
     "moderator-tools": "Moderator Tools",
     "moderation-header": "Moderation",
-    "hide-deleted": "Hide Deleted"
+    "hide-deleted": "Hide Deleted",
+    "view-discussions-log": "View Discussions Log"
```
Comment (jsutterfield) [inline front/common/public/locales/en/discussion.json]: Do we have an i18n ticket for this?

File: front/main/app/components/application-wrapper.js [hunk 6]
```diff
@@ -42,6 +42,11 @@ export default Component.extend(ResponsiveMixin, {
 		return `${vertical}-vertical`;
 	}),
 
+	viewChangeObserver: observer('responsive.isMobile', function () {
```
Comment (jsutterfield) [inline front/main/app/components/application-wrapper.js]: qq, will this be re-registered when the the window changes in the other direction?

Comment (katanka) [inline front/main/app/components/application-wrapper.js]: discussed offline

File: front/main/app/components/application-wrapper.js (hunk 4 above)
Comment (katanka) [inline front/main/app/components/application-wrapper.js]: Cool, I didn't make a decision between the two but I agree!

File: front/common/public/locales/en/discussion.json (hunk 5 above)
Comment (katanka) [inline front/common/public/locales/en/discussion.json]: We do now!

**Pull request merged**